MultiplayerTestRunner/src/SpaceRAII.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "CSP/Systems/Spaces/SpaceSystem.h"
+
+#include <optional>
+#include <string>
+
+/*
+ * RAII container object to facilitate automatically cleaning up a space when leaving scope
+ * This involves 2 actions, leaving the space, and destroying the space.
+ * This type will create a new random space upon construction if a SpaceId is not set.
+ * If a SpaceID is set, this type only joins and leaves the room, not destroying it, as it is assumed to already exist.
+ */
+class SpaceRAII
```

Comment:
[consider] Name could be "ScopedSpace"?

Reply:
Because RAII is such a terrible name for the concept? It is at that, but I think I prefer it here, just because `Scope` can be interpreted in other ways, where this object very specifically exists in order to leverage its destructor to do cleanup.